### PR TITLE
added method: search

### PR DESCRIPTION
**Pull request by evgeniusorg**

### CODE REVIEW

File: lib/confluence.js [hunk 1]
```diff
@@ -336,4 +336,77 @@ Confluence.prototype.updateAttachmentData = function(space, id, attachmentId, fi
         });
 };
 
+/**
+ *  This allows you to search by query
+ *
+ * @param {string} query
+ * @param {Function} callback
+ */
+Confluence.prototype.search = function(query, callback){
```
Comment (johnpduane) [inline lib/confluence.js:345]: The search function looks like a great addition. Thanks!

File: lib/confluence.js [hunk 2]
```diff
@@ -336,4 +336,77 @@ Confluence.prototype.updateAttachmentData = function(space, id, attachmentId, fi
...
+ * Put/update data in labels a page.
+ *
+ * @param {string} space
+ * @param {string} id
+ * @param {number} version
+ * @param {string} title
+ * @param {array} content
+ * @param {Function} callback
+ */
+Confluence.prototype.putLabel = function(space, id, version, title, content, callback){
```
Comment (johnpduane) [inline lib/confluence.js:366]: I think we should revise this implementation to use confluence's [content/{id}/label](https://docs.atlassian.com/atlassian-confluence/REST/latest/#content/{id}/label) endpoint for this. Do you agree? If so, we can add the other verbs at some point.

File: lib/confluence.js [hunk 3]
```diff
@@ -336,4 +336,77 @@ Confluence.prototype.updateAttachmentData = function(space, id, attachmentId, fi
...
+};
+
+/**
+ *  This allows you to search in plugins for query 
+ *
+ * @param {string} path
+ * @param {string} query
+ * @param {Function} callback
+ */
+Confluence.prototype.pluginsSearch = function(path, query, callback){
```
Comment (johnpduane) [inline lib/confluence.js:401]: Do we really need this function? In your test, you use `/rest/api/content/` as the path and embed a page Id in the query so this looks functionally equivalent to using `getContentById` or `getCustomContentById` but without the support for older versions of the api. Do you agree?  If so, let's remove this function and its test.

**Review by johnpduane — COMMENTED** [review]
Hi @evgeniusorg -

Sorry for the delay looking at this PR and thanks for submitting it.

I embedded a few comments for each of the new functions. Can you take a look and revise the pull request based on them?

Thanks for contributing!
John.

File: lib/confluence.js (hunk 1 above)
Comment (KenHollandWHY) [reply lib/confluence.js:345]: @johnpduane 
Hi John, I see that the search function has been approved but I dont see it in the master branch, is it possible to add it in?

**Comment by johnpduane** [comment]
The search method was integrated into v1.2.1. If there is still interest in the other methods in this PR, let's talk through my comments above. Thanks for adding search to the mix!

**Pull request closed without merging**